File: scripts/split_prompts.py

import json
import os

# Read the original file
with open('data/prompts.json', 'r', encoding='utf-8') as f:
    data = json.load(f)

print(f"Total items: {len(data)}")

# Split into chunks of 50 to be safe with context window
chunk_size = 50
chunks = [data[i:i + chunk_size] for i in range(0, len(data), chunk_size)]

for i, chunk in enumerate(chunks):
    filename = f'data/temp_chunk_{i}.json'
    with open(filename, 'w', encoding='utf-8') as f:
        json.dump(chunk, f, ensure_ascii=False, indent=2)
    print(f"Created {filename} with {len(chunk)} items")

File: scripts/split_prompts.js

const fs = require('fs');
const path = require('path');

try {
  const data = JSON.parse(fs.readFileSync('data/prompts.json', 'utf8'));
  console.log(`Total items: ${data.length}`);

  const chunkSize = 50;
  let chunkIndex = 0;
  
  for (let i = 0; i < data.length; i += chunkSize) {
    const chunk = data.slice(i, i + chunkSize);
    const filename = `data/temp_chunk_${chunkIndex}.json`;
    fs.writeFileSync(filename, JSON.stringify(chunk, null, 2), 'utf8');
    console.log(`Created ${filename} with ${chunk.length} items`);
    chunkIndex++;
  }
} catch (error) {
  console.error('Error:', error);
}

